lsp-mode.el

```diff
@@ -4519,7 +4519,12 @@ The method uses `replace-buffer-contents'."
                  (seq-sort #'lsp--text-edit-sort-predicate)
                  (mapc (lambda (edit)
                          (progress-reporter-update reporter (cl-incf done))
-                         (funcall apply-edit edit))))
+                         (funcall apply-edit edit)
+                         (when (lsp-snippet-text-edit? edit)
+                           (-when-let ((&SnippetTextEdit :range (&RangeToPoint :start)
+                                                         :insert-text-format? :new-text) edit)
+                             (when (eq insert-text-format? lsp/insert-text-format-snippet)
+                               (yas-expand-snippet (lsp--to-yasnippet-snippet new-text) start (+ start (length new-text)))))))))
```

Comment:
You may need to wrap this one with `let (yas-indent-line)` too. To prevent snippet indentation.

Reply:
Fixed.